### ISSUE TITLE
`datetime.display` long representations do not take the text language into consideration

### ISSUE THREAD

**Issue by chloe-the-catgirl** [typst/typst#1537]
### Description

> I would let it depend on whatever is set in `text`. So for example if someone sets `#set text(lang: "de")`, the date should use the German version as well.
> _Originally posted by @LaurenzV in https://github.com/typst/typst/issues/303#issuecomment-1489005819_

This is not working right now. Minimal working example:

```typst
#set text(lang: "de")
#datetime.today().display("[month repr:long]")

// Output is: June
```

For me, it's not critical, just wanted to write it down, so it does not get forgotten.

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by laurmaedje** [comment]
Yes, that's a known limitation. The formatting to string happens at a stage where the text language is not yet known. Hopefully we can improve this in the future.

**Comment by LaurenzV** [comment]
Even if we would know the text language, it's still going to be tricky because the underlying library we use doesn't support  translating the formatted months/weeks into a different language. Will probably require us to somehow implement this on our side.

**Comment by GoldsteinE** [comment]
There needs to be some way to specify declension for this to work with some languages, e.g. it’s «13 ноября», not «13 ноябрь».

TeX’s datetime2 does this by redefining date display comments, which is IMO not ideal.

https://www.dickimaw-books.com/faq.php?action=view&category=datetime2#datedeclension

**Comment by FNG-2002** [comment]
Hello everyone, is there any update on this?

**Comment by laurmaedje** [comment]
This is currently blocked on larger work to enable show rules on types like `datetime`, so it will still take some time.

**Comment by LaurenzV** [comment]
FWIW there is currently a package that allows you to format datetimes using different locales: https://typst.app/universe/package/icu-datetime

Perhaps we should also rework the Typst implementation to use `icu4x` instead of the `time` crate for date formatting. The main challenge would probably be how to lazily load the `icu4x` data.

**Comment by laurmaedje** [comment]
How much data are we talking?

**Comment by LaurenzV** [comment]
I'm not sure, but according to the package it's 8MB... Although I guess some of it will also be the wasm code for the library.

**Comment by laurmaedje** [comment]
Okay that's a lot.

**Comment by LaurenzV** [comment]
But from what I've seen icu data should be pretty modular, so it should be possible to load whatever you need on demand. But I haven't looked into it more deeply.
